doc/design/katib_on_sqlflow.md

```diff
@@ -1,70 +1,91 @@
+1. Argo manages the status of workflow executions.
 
-In SQLFlow, Couler codegen will generate a python program from the input SQL statement. In the generated python program, it invokes Couler APIs (`couler.run_container(...)`) to create Argo steps. However, the current generated python program only calls `couler.run_container` to create a single container step, which does not work with job type steps (e.g., Katib). The container step runs a single container while a job step may launch multiple containers which will coordinate with each other. 
+Argo takes workflows in the form of YAML files, and it is error-prone to write such YAML files manually.  So, we created [Couler](/python/couler/README.md) as an intermediate programmatic representation of workflows.
 
-Then it needs to invoke different Couler APIs according to input SQL queries. In Katib case, Couler codegen needs to check if the input SQL query is to create a Katib job. If it is, the codegen should generate codes to invoke a Couler Katib API, like `couler.run_katib(...)`
+We need to develop a new codegen, `codegen_couler.go`, for SQLFlow.  `codegen_couler.go` converts the parsed SQL program, a.k.a., the [intermediate representation](/pkg/sql/ir), or IR, into a Couler program.
 
-## Design for `couler.run_katib(...)`
+### The Integration via Couler
 
-Considering Katib itself supports multiple models and frameworks, and more may come in the future, we use a unique API to create Katib steps in Couler. The API is as following:
+SQLFlow parses each SQL program into an IR, which is a list of statement IRs.  The `codegen_couler.go` converts the IR into a Couler program.   We need to add a Couler functions `couler.katib.train` for the calling by the generated Couler program.
 
-`couler.run_katib("model_or_framework"=None, katib_params={}, model_or_framework_params={})`
+Consider the following example program.
 
-In this API, it includes three arguments:
-- `model_or_framework`: string, indicates the model (e.g., XGBoost) or frameworks (e.g., tf or pytorch) used in model training on Katib.
-- `katib_params`: dict, configures Katib jobs. For example, users can specify max trials in model training, like `katib_params= { "max_trial_count": 10}`  
-- `model_or_framework_params`: dict, configures model or framework given in `model_or_framework`. The above XGBoost example will be: `model_or_framework_params= { "booster": "gbtree", "objective": "multi:softmax", "eta": 1, "num_round": [20, 100], max_depth: [] }`
+```sql
+SELECT * FROM a, b WHERE a.id = b.id INTO c;
+SELECT * FROM c TO TRAIN model_def WITH objective=multi:softmax, eta=1 LABEL class INTO my_xgb_model;
+```
+
+The `codegen_couler.go` might generate the following Couler program.
+
+```python
+couler.maxcompute.run("""SELECT * FROM a, b WHERE a.id = b.id INTO c;
+                         SELECT * FROM c INTO temp""")
+couler.maxcompute.export(table="temp", file="/hdfs/temp")
```

Comment:
It seems this `temp` is unnecessary.

Reply:
Oh. I mean to export the table temp to be a file on HDFS named temp. So, I think we need both "temp", am I right?

Reply:
Could we export the table `c` to the HDFS file？It seems `temp` is identical to `c`?

Reply:
"INTO temp" means saving the output from the SELECT operation into a new table named temp.